nbextensions/bulkImportCell/tabs/xsvGenerator.js

```diff
@@ -0,0 +1,250 @@
+                    p('Destination directory'),
+                    div({
+                        id: this.id,
+                    }),
+                ]
+            );
+        }
+
+        /**
+         *  Coerce the form and file upload params into the appropriate format
+         *  for the staging service endpoint
+         *
+         * @param {object} formParams validated parameters from the form with keys
+         *      output_file_type: format to output the file in
+         *      output_directory: where to save the file
+         *      types: validated array of input types to include in the request
+         * @returns {object} output, suitable for sending to the staging service endpoint
+         */
+        generateRequest(formParams) {
+            const { output_file_type, output_directory, types } = formParams;
+
+            const params = this.model.getItem('params'),
+                appSpecs = this.model.getItem('app.specs'),
+                output = {
+                    output_directory,
+                    output_file_type,
+                    types: {},
+                };
+
+            types.forEach((fileType) => {
+                const appId = this.typesToFiles[fileType].appId;
+                output.types[fileType] = {
+                    order_and_display: appSpecs[appId].parameters.map((param) => {
+                        return [param.id, param.ui_name];
+                    }),
+                    data: params[fileType].filePaths.map((element) => {
+                        const datum = JSON.parse(JSON.stringify(element));
+                        Object.keys(params[fileType].params).forEach((param) => {
+                            datum[param] = params[fileType].params[param];
+                        });
+                        return datum;
+                    }),
+                };
+            });
+            return output;
+        }
+
+        /**
+         * Send the request to the staging service client
+         * @param {object} requestParams the body for the request
+         * @returns jQuery.ajax
+         */
+        sendRequest(requestParams) {
+            if (!this.stagingServiceClient) {
+                this.stagingServiceClient = new StagingServiceClient({
+                    root: this.runtime.config('services.staging_api_url.url'),
+                    token: this.runtime.authToken(),
+                });
+            }
+            return this.stagingServiceClient.write_bulk_specification({
+                data: requestParams,
+                dataType: 'json',
+                headers: { 'Content-Type': 'application/json' },
+            });
+        }
+
+        /**
+         *  Display the results of the query to generate the template
+         *
+         * @param {object} resp response from the staging service endpoint
+         */
+        displayResult(resp) {
+            if (resp.error || !resp.files_created || !Object.keys(resp.files_created).length) {
```

Comment:
unfortunately the endpoint will return a successful response if no files were created, so you have to check to ensure that there are some results to show!